capstone/src/main/webapp/youtubeScript.js

```diff
@@ -64,24 +86,26 @@ async function inputCommentsToPerspective(comments) {
   if (!requestedAttributes) {
       return;
   }
-  const attributeScores = [];
-  for (const item in comments.items) {
-    const perspectiveScore = await callPerspective(comments.items[item].snippet.topLevelComment.snippet.textOriginal, langElement.value, requestedAttributes);
-    attributeScores.push(perspectiveScore);
-  }
   const attributeTotals = new Map();
-  for (var i = 0; i < requestedAttributes.length; i++) {
-    for (var j = 0; j < attributeScores.length; j++) {
-      if (attributeTotals.has(requestedAttributes[i])) {
-        attributeTotals.set(requestedAttributes[i], attributeTotals.get(requestedAttributes[i]) + attributeScores[j].attributeScores[requestedAttributes[i]].summaryScore.value);
-      } else {
-        attributeTotals.set(requestedAttributes[i], attributeScores[j].attributeScores[requestedAttributes[i]].summaryScore.value);
+  for (const comments in commentsList) {
```

Comment:
Could we put these average calculations into a separate helper method to make this one a little more readable?

Reply:
addressed in "separate into classes and utilize promise"